### PR TITLE
Redirect user back to landing page

### PR DESCRIPTION
**Pull request by dukex**
## Proposed Changes

Using oAuth login, we are not able back to landed page after login.

The steps to reproduce the problem:

1. Set up the a oauth to rabbitmq_management 
1. Open `https://localhost:15672/#/queues/events/my-queue`
2. You will see the login page
3. Click on "Click here to log in" 
4. Log in your identify provider
5. You will be redirected to `https://localhost:15672/`

To create a good user-experience would nice redirect the user to previous landed page, in my example, this page was `https://localhost:15672/#/queues/events/my-queue`

This pull request add this feature, now we save the landed page and after login we redirect back the user to landed page.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

### CODE REVIEW

File: deps/rabbitmq_management/priv/www/js/main.js [hunk 1]
```diff
@@ -7,6 +7,8 @@ $(document).ready(function() {
      renderWarningMessageInLoginStatus(fmt_escape_html(error));
    } else {
       if (oauth.enabled) {
+        store_pref("oauth-return-to", url.hash);
```
Comment (MarcialRosales) [inline deps/rabbitmq_management/priv/www/js/main.js:10]: Maybe we can move `store_pref("oauth-return-to", url.hash);` inside `startWithOAuthLogin()`?

File: deps/rabbitmq_management/selenium/test/oauth/with-idp-initiated/redirection-after-login.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const { By, Key, until, Builder } = require('selenium-webdriver')
+require('chromedriver')
+const assert = require('assert')
+const { buildDriver, goToExchanges, captureScreensFor, teardown, goToHome } = require('../../utils')
+
+const SSOHomePage = require('../../pageobjects/SSOHomePage')
+const UAALoginPage = require('../../pageobjects/UAALoginPage')
+const ExchangesPage = require('../../pageobjects/ExchangesPage')
+const OverviewPage = require('../../pageobjects/OverviewPage')
```
Comment (MarcialRosales) [inline deps/rabbitmq_management/selenium/test/oauth/with-idp-initiated/redirection-after-login.js:9]: It is cosmetic, but given that Overview is not used in this test , we do not need to declare it,

File: deps/rabbitmq_management/selenium/test/oauth/with-sp-initiated/redirection-after-login.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+const { By, Key, until, Builder } = require('selenium-webdriver')
+require('chromedriver')
+const assert = require('assert')
+const { buildDriver, goToExchanges, captureScreensFor, teardown, goToHome } = require('../../utils')
+
+const SSOHomePage = require('../../pageobjects/SSOHomePage')
+const UAALoginPage = require('../../pageobjects/UAALoginPage')
+const ExchangesPage = require('../../pageobjects/ExchangesPage')
+const OverviewPage = require('../../pageobjects/OverviewPage')
```
Comment (MarcialRosales) [inline deps/rabbitmq_management/selenium/test/oauth/with-sp-initiated/redirection-after-login.js:9]: Same comment for Overview page

**Review by MarcialRosales — CHANGES_REQUESTED** [review]
Thank you very much for this missing feature. It was in the TODO list.

File: deps/rabbitmq_management/selenium/test/oauth/with-idp-initiated/redirection-after-login.js (hunk 2 above)
Comment (dukex) [reply deps/rabbitmq_management/selenium/test/oauth/with-idp-initiated/redirection-after-login.js]: Oh sorry, I was creating a testing using the overview page, but I changed it later and forgot to remove here, thanks

File: deps/rabbitmq_management/priv/www/js/main.js (hunk 1 above)
Comment (dukex) [reply deps/rabbitmq_management/priv/www/js/main.js]: Done, thanks.

**Comment by dukex** [comment]
@MarcialRosales thanks for your review, can you please review it again?

**Review by MarcialRosales — APPROVED** [review]

**Comment by MarcialRosales** [comment]
@dukex I am further investigating the changes as I have noticed the new test cases have failed in the CI.

**Comment by MarcialRosales** [comment]
@dukex do you mind if I rebase your PR to take the latest changes that went onto `main` branch?

**Comment by dukex** [comment]
@MarcialRosales don't worry, you can rebase it :D

**Comment by michaelklishin** [comment]
@dukex thank you for contributing to RabbitMQ!

**Pull request merged**